### PR TITLE
Update various dependencies

### PR DESCRIPTION
**Pull request by kordianbruck**
Fix breakages with new calendar version. Update to minSDK 24 because QR code scanner updated.

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt [hunk 1]
```diff
@@ -116,7 +114,7 @@ class CalendarFragment : FragmentForAccessingTumOnline<EventsResponse>(
                 todayButton.visibility = if (isToday) View.GONE else View.VISIBLE
                 TransitionManager.beginDelayedTransition(swipeRefreshLayout)
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt:115]: I'm not even sure if this is needed anymore. Ideas on replacing this piece of code?

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesFragment.kt [hunk 2]
```diff
@@ -161,7 +161,7 @@ class GradesFragment : FragmentForAccessingTumOnline<ExamList>(
 
             xAxis.apply {
                 granularity = 1f
-                setValueFormatter { value, _ -> grades[value.toInt()] }
+                //setValueFormatter { value, _ -> grades[value.toInt()] }
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesFragment.kt:164]: It seems to be working just fine without this formatter.

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt (hunk 1 above)
Comment (Bentipa) [reply app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt:116]: Do you know what this was for? Showing a separate "Today"-Button but hide it when the day is visible?

Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt:116]: Yes, it was hidden before. I've actually updated it now to always show that button. I don't think its so super useful to hide the button.

**Pull request merged**